extensions/amp-story/0.1/amp-story.js

```diff
@@ -844,46 +843,73 @@ export class AmpStory extends AMP.BaseElement {
 
     this.element.appendChild(this.bookend_.build());
 
-    this.loadRelatedArticles_().then(articleSets => {
-      if (articleSets === null) {
-        return;
+    this.setAsOwner(this.bookend_.getRoot());
+
+    this.loadBookendConfig_().then(bookendConfig => {
+      if (bookendConfig !== null) {
+        this.bookend_.setConfig(dev().assert(bookendConfig));
       }
-      this.bookend_.setRelatedArticles(dev().assert(articleSets));
+      this.scheduleResume(this.bookend_.getRoot());
     });
   }
 
 
   /**
-   * @return {!Promise<?Array<!./related-articles.RelatedArticleSet>>}
+   * @return {!Promise<?./bookend.BookendConfig>}
    * @private
    */
-  loadRelatedArticlesImpl_() {
-    const rawUrl = this.getRelatedArticlesUrlOptional_();
+  loadBookendConfigImpl_() {
+    // two-tiered implementation for backwards-compatibility with
+    // related-articles attribute
+    return this.loadBookendConfigInternal_().then(bookendConfig =>
+        bookendConfig || this.loadRelatedArticlesAsBookendConfig_());
+  }
+
+
+  /**
+   * @return {!Promise<?./bookend.BookendConfig>}
+   */
+  loadBookendConfigInternal_() {
+    return this.loadJsonFromAttribute_(BOOKEND_CONFIG_ATTRIBUTE_NAME)
+        .then(response => response && {
+          shareProviders: response['share-providers'],
+          relatedArticles: response['related-articles'] ?
+              buildFromJson(response['related-articles']) : [],
+        });
+  }
+
 
-    if (rawUrl === null) {
+  /**
+   * @return {!Promise<?./bookend.BookendConfig>}
+   * @private
+   */
+  loadRelatedArticlesAsBookendConfig_() {
+    return this.loadJsonFromAttribute_(RELATED_ARTICLES_ATTRIBUTE_NAME)
```

Comment:
Just to double check to make sure I'm reading this right: we'll still support the old bookend JSON, but also support the new one?

Reply:
I was thinking we could deprecate it for a couple of weeks and remove it then.